sync-core/src/main/java/com/cloudant/mazha/CouchConfig.java

```diff
@@ -144,6 +151,23 @@ public void setMaxConnections(int maxConnections) {
 		this.maxConnections = maxConnections;
 	}
 
+    public Map<String, String> getCustomHeaders() {
+        return customHeaders;
+    }
+
+    public void setCustomHeaders(Map<String, String> customHeaders) {
+        this.customHeaders = new HashMap<String, String>();
+        List<String> prohibitedHeaders = Arrays.asList("authorization", "www-authenticate",
+                "host", "connection", "content-type", "accept", "content-length");
+        for (Map.Entry<String, String> header : customHeaders.entrySet()) {
+            if (prohibitedHeaders.contains(header.getKey().toLowerCase())) {
+                throw new IllegalArgumentException("Bad optional HTTP header: " + header);
```

Comment:
I think we should have a better exception message, it should state that the header is not allowed to be set by the user